tests/mesh/test-aabb-helpers.cpp

```diff
@@ -0,0 +1,162 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+/*
+ * Modified from RINGMesh https://github.com/ringmesh/RINGMesh
+ * Copyright (c) 2012-2018, Association Scientifique pour la Geologie et ses
+ * Applications (ASGA)
+ */
+
+#include <geode/basic/logger.h>
+
+#include <geode/mesh/builder/triangulated_surface_builder.h>
+#include <geode/mesh/core/triangulated_surface.h>
+
+#include <geode/mesh/core/aabb_helpers.h>
+
+#include <geode/tests/common.h>
+
+using namespace geode;
+
+template < index_t dimension >
+Point< dimension > create_vertex( double i, double j )
+{
+    Point< dimension > point;
+    point.set_value( 0, i );
+    point.set_value( 1, j );
+    return point;
+}
+
+template < index_t dimension >
+void add_vertices(
+    TriangulatedSurfaceBuilder< dimension >& builder, index_t size )
+{
+    builder.create_vertices( size * size );
+    index_t id = 0;
+    for( index_t i : Range( size ) )
+    {
+        for( index_t j : Range( size ) )
+        {
+            builder.set_point( id++, create_vertex< dimension >( i, j ) );
+        }
+    }
+}
+
+template < index_t dimension >
+void add_triangles(
+    TriangulatedSurfaceBuilder< dimension >& builder, index_t size )
+{
+    builder.create_triangles( ( size - 1 ) * ( size - 1 ) * 2 );
+    index_t id = 0;
+    for( index_t i : Range( size - 1 ) )
+    {
+        for( index_t j : Range( size - 1 ) )
+        {
+            builder.set_polygon_vertex( { id, 0 }, i * size + j );
+            builder.set_polygon_vertex( { id, 1 }, i * size + j + 1 );
+            builder.set_polygon_vertex( { id, 2 }, ( i + 1 ) * size + j );
+            id++;
+            builder.set_polygon_vertex( { id, 0 }, i * size + j + 1 );
+            builder.set_polygon_vertex( { id, 1 }, ( i + 1 ) * size + j + 1 );
+            builder.set_polygon_vertex( { id, 2 }, ( i + 1 ) * size + j );
+            id++;
+        }
+    }
+}
+
+template < index_t dimension >
+void check_tree( const AABBTree< dimension >& tree,
+    const DistanceToTriangle< dimension >& distance_action,
+    index_t size )
+{
+    double offset = 0.2;
+    index_t id = 0;
+    for( index_t i : Range( size - 1 ) )
+    {
+        for( index_t j : Range( size - 1 ) )
+        {
+            Point< dimension > query1 =
```

Comment:
```suggestion
            const auto query1 =
```
prefer always `const auto` if possible

Reply:
Could you do it for the entire file?